src/js/containers/search/table/ResultsTableContainer.jsx

```diff
@@ -132,10 +132,10 @@ export class ResultsTableContainer extends React.Component {
         // load every possible table column up front, so we don't need to deal with this when
         // switching tabs
         const columns = tableTypes.concat(subTypes).reduce((cols, type) => {
-            const visibleColumns = defaultColumns(type.internal);
+            const visibleColumns = map(defaultColumns(type.internal), (data) => data.title);
             const parsedColumns = {};
-            visibleColumns.forEach((title) => {
-                parsedColumns[title] = this.createColumn(title);
+            defaultColumns(type.internal).forEach((data) => {
+                parsedColumns[data.title] = this.createColumn(data.displayName, data.title);
```

Comment:
I think we should use es6 methods as much as possible rather than lodash, in this case then we could turn line 137 to a `.reduce` and line 135 to use `.map`

```javascript
// line 135
visibleColumns = defaultColumns(type.internal).map((col) => col.title);
// line 136 + 137
parsedColumns = defaultColumns(type.internal).reduce((parsedCols, data) => {
    return Object.assign({}, parsedCols, {
        [data.title]: this.createColumn(data.displayName, data.title)
    });
}, {});
```

Reply:
Couldnt line 138 also be   `this.createColumn(data.displayName || data.title)`? That way we could just use one param in this.createColumn

Reply:
Yeah, I do not really care JS map or reduce v. Lodash map or reduce to simply map or reduce an array. The reasons I use Lodash are because I used it heavily on my last project so I am extremely comfortable with it. We use it in a lot of places in this code base. There are a lot of useful functions that Lodash has already, that I also could write myself with JS map and JS reduce, and other JS methods, but since Lodash already has them, this saves me time and provides standardization without having to create our own.

The `forEach` for `parsedColumns` is a good catch, but we should just use map. Not sure why to use reduce?

The `createColumn` function needs two fields because it is setting the `columnName` and a `displayName` property among others for the column data set. The `columnName` is more of a key for the data sent to the API and the `displayName` is what the user sees.  When clicking sort and probably other functionality the table sends the `columnName`. If a column had a `displayName` and we set the `columnName` to the `displayName` we would get an error from the API.

Reply:
@jonhill13 ,

You're returning an object in the `.forEach` loop. In ES6 `.map` always returns an array, so in this case that's why you'd want to use `.reduce`. I'm good w/ you using `lodash` but I'm gonna use `ES6`, maybe this way we can both introduce one another to the various benefits of each. I personally don't have a ton of experience with lodash and would definitely be interested in learning while I review your PRs.

Understood regarding the `createColumn` fn. Thanks for the clarification, I see that now you are using only `title` for the `columnName` property.

Reply:
Yeah, I like that using reduce to return and object instead of creating one before the forEach.

Reply:
👌